client/src/featureform/client.py

```diff
@@ -0,0 +1,75 @@
+from typing import Union
+from .register import (
+    ResourceClient,
+    SourceRegistrar,
+    LocalSource,
+    SubscriptableTransformation,
+)
+from .serving import ServingClient
+
+
+class Client(ResourceClient, ServingClient):
+    """
+    Client for interacting with Featureform APIs (resources and serving)
+
+    **Using the Client:**
+    ```py title="definitions.py"
+    import featureform as ff
+    from featureform import Client
+
+    client = Client("http://localhost:8080")
+
+    # Example 1: Get a registered provider
+    redis = client.get_provider("redis-quickstart")
+
+    # Example 2: Compute a dataframe from a registered source
+    transactions_df = client.compute_df("transactions", "quickstart")
+    """
+
+    def __init__(
+        self, host=None, local=False, insecure=False, cert_path=None, dry_run=False
+    ):
+        ResourceClient.__init__(
+            self,
+            host=host,
+            local=local,
+            insecure=insecure,
+            cert_path=cert_path,
+            dry_run=dry_run,
+        )
+        # Given both ResourceClient and ServingClient are instantiated together, if dry_run is True, then
+        # the ServingClient cannot be instantiated due to a conflict the local and host arguments.
+        if not dry_run:
+            ServingClient.__init__(
```

Comment:
super(ServingClient, self). __init

Can you explain why we can't do this if its a dry run? I don't understand the comment. We should also call super __init__ whenever possible

Reply:
The reason being:
If you were to cast this as a ServingClient and use it, you'd end up with some weird behavior if dry run is set

Reply:
Currently, the logic in `ServingClient`'s constructor will throw an exception is `local` is `False` and `host` is `None`, which is true in the case when a user calls `> featureform apply definitions.py --dry-run`. Given the presence of `dry_run` indicates that nothing can be done with `ServingClient`, I followed the pattern I saw in `ResouceClient` and put an early return here to avoid the unindented exception.

Reply:
Dry run was a kind of hacky thing I added to test things out a while ago. I think we should have a larger conversation outside of this task about what dry run should actually do and how we want to support it